python/lsst/ip/isr/ptcDataset.py

```diff
@@ -0,0 +1,472 @@
+            calib.finalMeans.setdefault(ampName, amp['finalMeans'])
+            calib.photoCharge.setdefault(ampName, amp['photoCharge'])
+
+        calib.updateMetadata()
+        return calib
+
+    def toDict(self):
+        """Return a dictionary containing the calibration properties.
+        The dictionary should be able to be round-tripped through
+        `fromDict`.
+        Returns
+        -------
+        dictionary : `dict`
+            Dictionary of properties.
+        """
+        self.updateMetadata()
+
+        outDict = {}
+        metadata = self.getMetadata()
+        outDict['metadata'] = metadata
+
+        outDict['ptcFitType'] = self.ptcFitType
+        outDict['ampNames'] = self.ampNames
+        outDict['badAmps'] = self.badAmps
+        outDict['inputExpIdPairs'] = self.inputExpIdPairs
+        outDict['expIdMask'] = self.expIdMask
+        outDict['rawExpTimes'] = self.rawExpTimes
+        outDict['rawMeans'] = self.rawMeans
+        outDict['rawVars'] = self.rawVars
+        outDict['gain'] = self.gain
+        outDict['gainErr'] = self.gainErr
+        outDict['noise'] = self.noise
+        outDict['noiseErr'] = self.noiseErr
+        outDict['ptcFitPars'] = self.ptcFitPars
+        outDict['ptcFitParsError'] = self.ptcFitParsError
+        outDict['ptcFitChiSq'] = self.ptcFitChiSq
+        outDict['covariances'] = self.covariances
+        outDict['covariancesModel'] = self.covariancesModel
+        outDict['covariancesSqrtWeights'] = self.covariancesSqrtWeights
+        outDict['aMatrix'] = self.aMatrix
+        outDict['bMatrix'] = self.bMatrix
+        outDict['covariancesNoB'] = self.covariancesNoB
+        outDict['covariancesModelNoB'] = self.covariancesModelNoB
+        outDict['covariancesSqrtWeightsNoB'] = self.covariancesSqrtWeightsNoB
+        outDict['aMatrixNoB'] = self.aMatrixNoB
+        outDict['finalVars'] = self.finalVars
+        outDict['finalModelVars'] = self.finalModelVars
+        outDict['finalMeans'] = self.finalMeans
+        outDict['photoCharge'] = self.photoCharge
+
+        return outDict
+
+    @classmethod
+    def fromTable(cls, tableList):
+        """Construct calibration from a list of tables.
+        This method uses the `fromDict` method to create the
+        calibration, after constructing an appropriate dictionary from
+        the input tables.
+        Parameters
+        ----------
+        tableList : `list` [`lsst.afw.table.Table`]
+            List of tables to use to construct the datasetPtc.
+        Returns
+        -------
+        calib : `lsst.cp.pipe.`
+            The calibration defined in the tables.
+        """
+        ptcTable = tableList[0]
+
+        metadata = ptcTable.meta
+        inDict = dict()
+        inDict['metadata'] = metadata
+        inDict['ampNames'] = dict()
+
+        for record in ptcTable:
+            ampName = record['AMPLIFIER_NAME']
+
+            inDict['ampNames'][ampName] = {
```

Comment:
This seems to be different than the format in `fromDict`.  Am I just misreading one of these methods?

Reply:
I changed it so that it is similar to the format in `fromDict`